src/Compilers/CSharp/Test/Semantic/Semantics/InterpolationTests.cs

```diff
@@ -13832,23 +13832,71 @@ public void RefEscape_11(LanguageVersion languageVersion)
 }
 class Program
 {
-    static void Main()
+    static void F1()
     {
         Span<char> s = stackalloc char[10];
         M($""{s}"");
     }
+    static void F2()
+    {
+        Span<char> s = stackalloc char[10];
+        CustomHandler h2 = new CustomHandler(0, 1);
+        h2.AppendFormatted(s); // 1
```

Comment:
I didn't understand why the interpolated string version had no errors but the "explicit" version has this error. Could you please clarify?

Reply:
My understanding is the interpolated string version is considered as creating a `CustomHandler` instance as if the `new CustomHandler()` and the interpolated string arguments were combined as an object initializer. In `F1()`, where the interpolated string argument `s` has safe-to-escape of the current method, the resulting `CustomHandler` instance will have safe-to-escape of the current method as well. In `F2()`, the argument `s` is not included in the `CustomHandler` object initializer, so the resulting instance is safe-to-escape to the caller.

Reply:
> My understanding is the interpolated string version is considered as creating a CustomHandler instance as if the new CustomHandler() and the interpolated string arguments were combined as an object initializer.

I would expect these scenarios to have the same behavior.

Reply:
> I would expect these scenarios to have the same behavior.

If we make the `CustomHandler` in the first example explicit, the two examples are essentially:
```csharp
CustomHandler h1 = $"{s}";
M(h1);

CustomHandler h2 = new CustomHandler(0, 1);
h2.AppendFormatted(s);
M(h2);
```

The difference between the two is the initializers: the initializer for `h1` limits the scope to the current method; the initializer for `h2` does not limit the scope.